### PR TITLE
added API to fetch users from ES

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: controllers/v1/entities.js [hunk 1]
```diff
@@ -319,5 +319,60 @@ module.exports = class Entities extends Abstract {
...
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
+  * @apiSampleRequest /kendra/api/v1/entities/getUsersByEntityAndRole
+  * @apiParamExample {json} Request-body:
+  * {
+    "entityId" : "5da829874c67d63cca1bd9d0",
+    "role" : "HM"
+  }
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
```
Comment (aks30) [inline controllers/v1/entities.js:331]: @deepakr123 - Sample response is missing

File: module/entities/validator/v1.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ module.exports = (req) => {
         },
         subEntitiesRoles : function() {
             req.checkParams('_id').exists().withMessage("required Entity id");
+        },
+        getUsersByEntityAndRole: function () {
+            req.checkBody('entityId').exists().withMessage("entityId id is required");
```
Comment (aks30) [inline module/entities/validator/v1.js]: @deepakr123 - Add valid mongoID validation

File: controllers/v1/entities.js [hunk 3]
```diff
@@ -319,5 +319,60 @@ module.exports = class Entities extends Abstract {
...
+   * Get users by entityId and role.
+   * @method
+   * @name getUsersByEntityAndRole
+   * @param {Object} req - request data.
+   * @param {String} req.body.entityId - entity id.
+   * @param {String} req.body.role - role code.
+   * @returns {JSON} Array of entities and users.
+   */
+
+  getUsersByEntityAndRole(req) {
```
Comment (aks30) [inline controllers/v1/entities.js:360]: @deepakr123 - Would suggest making it a get call, params will be entityId and query parameter will be role code.

File: module/entities/helper.js [hunk 4]
```diff
@@ -595,4 +596,110 @@ module.exports = class EntitiesHelper {
...
+            entityMatch.match[`data.telemetry_entities.${entityDocument[0].entityType}_id`] = entityId;
+
+            queryObject.query.bool.must.push(entityMatch);
+           
+            let entityDocuments = await elasticSearch.searchDocumentFromIndex
+            (
+                process.env.ELASTICSEARCH_ENTITIES_INDEX,
+                "_doc",
+                queryObject,
+                null,
```
Comment (aks30) [inline module/entities/helper.js:661]: @deepakr123 - Do we need to pass null ? Just don't pass those parameters

File: module/entities/helper.js [hunk 5]
```diff
@@ -595,4 +596,110 @@ module.exports = class EntitiesHelper {
...
+                code : role
+            },[
+                "entityTypes"
+            ]);
+            
+            if(!userRoles.length) {
+                throw new Error(constants.apiResponses.INVALID_ROLE)
+            }
+
+            let queryObject = {
```
Comment (aks30) [inline module/entities/helper.js:647]: @deepakr123 - Let's change as per our discussion.

Fetch all sub-entities from MongoDB
For sub-entities get the doc from ES with a projection of only the required role.

File: controllers/v1/entities.js (hunk 3 above)
Comment (deepakr123) [reply controllers/v1/entities.js:360]: resolved

File: module/entities/validator/v1.js (hunk 2 above)
Comment (deepakr123) [reply module/entities/validator/v1.js]: resolved

File: module/entities/helper.js (hunk 4 above)
Comment (deepakr123) [reply module/entities/helper.js:661]: if we don't pass, default size 10 is being set which is used by some other API. SO i should pass null.

File: module/entities/helper.js (hunk 5 above)
Comment (deepakr123) [reply module/entities/helper.js:647]: resolved

File: controllers/v1/entities.js (hunk 1 above)
Comment (deepakr123) [reply controllers/v1/entities.js:331]: resolved

**Pull request closed without merging**